store/sqlstore/group_supplier.go

```diff
@@ -0,0 +1,864 @@
+			return result
+		}
+	} else {
+		member.DeleteAt = 0
+		var rowsChanged int64
+		var err error
+		if rowsChanged, err = s.GetMaster().Update(member); err != nil {
+			result.Err = model.NewAppError("SqlGroupStore.GroupCreateOrRestoreMember", "store.update_error", nil, "group_id="+member.GroupId+", user_id="+member.UserId+", "+err.Error(), http.StatusInternalServerError)
+			return result
+		}
+		if rowsChanged != 1 {
+			result.Err = model.NewAppError("SqlGroupStore.GroupCreateOrRestoreMember", "store.sql_group.no_rows_changed", nil, "", http.StatusInternalServerError)
+			return result
+		}
+	}
+
+	retrievedMember = nil
+
+	if err := s.GetMaster().SelectOne(&retrievedMember, "SELECT * FROM GroupMembers WHERE GroupId = :GroupId AND UserId = :UserId AND DeleteAt = 0", map[string]interface{}{"GroupId": member.GroupId, "UserId": member.UserId}); err != nil {
+		if err == sql.ErrNoRows {
+			result.Err = model.NewAppError("SqlGroupStore.GroupCreateOrRestoreMember", "store.sql_group.no_rows", nil, "group_id="+member.GroupId+"user_id="+member.UserId+","+err.Error(), http.StatusNotFound)
+		} else {
+			result.Err = model.NewAppError("SqlGroupStore.GroupCreateOrRestoreMember", "store.select_error", nil, "group_id="+member.GroupId+"user_id="+member.UserId+","+err.Error(), http.StatusInternalServerError)
+		}
+		return result
+	}
+
+	result.Data = retrievedMember
+	return result
+}
+
+func (s *SqlSupplier) GroupDeleteMember(ctx context.Context, groupID string, userID string, hints ...store.LayeredStoreHint) *store.LayeredStoreSupplierResult {
+	result := store.NewSupplierResult()
+
+	if !model.IsValidId(groupID) {
+		result.Err = model.NewAppError("SqlGroupStore.GroupDeleteMember", "model.group.id.app_error", nil, "", http.StatusBadRequest)
+		return result
+	}
+	if !model.IsValidId(userID) {
+		result.Err = model.NewAppError("SqlGroupStore.GroupDeleteMember", "model.group_member.user_id.app_error", nil, "", http.StatusBadRequest)
+		return result
+	}
+
+	var retrievedMember *model.GroupMember
+	if err := s.GetMaster().SelectOne(&retrievedMember, "SELECT * FROM GroupMembers WHERE GroupId = :GroupId AND UserId = :UserId AND DeleteAt = 0", map[string]interface{}{"GroupId": groupID, "UserId": userID}); err != nil {
+		if err == sql.ErrNoRows {
+			result.Err = model.NewAppError("SqlGroupStore.GroupDeleteMember", "store.sql_group.no_rows", nil, "group_id="+groupID+"user_id="+userID+","+err.Error(), http.StatusNotFound)
+			return result
+		}
+		result.Err = model.NewAppError("SqlGroupStore.GroupDeleteMember", "store.select_error", nil, "group_id="+groupID+"user_id="+userID+","+err.Error(), http.StatusInternalServerError)
+		return result
+	}
+
+	retrievedMember.DeleteAt = model.GetMillis()
+
+	if rowsChanged, err := s.GetMaster().Update(retrievedMember); err != nil {
```

Comment:
Seems simpler to just `UPDATE GroupMembers SET DeleteAt = :DeleteAt WHERE GroupId = :GroupId AND UserId = :UserId`. I'd also advocate for making deletion idempotent, vs. failing to delete a previously deleted record.